src/lib.rs

```diff
@@ -42,264 +46,186 @@ pub struct ShellCommandOutput {
 /// IE: 'sleep 9999' won't be killed cause no output is produced.
 /// Work-arounds/fixes are being considered.
 #[derive(Debug, Event)]
-pub struct KillShellCommand(pub u32);
+pub struct KillProcess(pub Pid);
+
+#[derive(Debug, Event, PartialEq, Eq, Clone)]
+pub enum ProcessError {
+    FailedToStart,
+}
 
 #[derive(Debug, Event)]
-pub struct ShellCommandCompleted {
+pub struct ProcessCompleted {
     pub success: bool,
     pub pid: u32,
     pub command: String,
-    pub output_buffer: Vec<String>,
 }
 
-struct ActiveShellCommand {
-    command: String,
-    task: Option<Task<(bool, Vec<String>)>>,
-    pid: u32,
-    output_lines: Option<Vec<String>>,
-    kill_requested: bool,
+/// The lines written to the standard output by a given process.
+#[derive(Debug, Default, Clone)]
+struct ProcessOutputBuffer(Arc<Mutex<Vec<String>>>);
+
+#[derive(Debug)]
+pub struct ActiveProcess {
+    command: Command,
+    process: Child,
+    task: Task<()>,
+    output_buffer: ProcessOutputBuffer,
 }
 
 #[derive(Default, Resource)]
-struct ActiveShellCommands(Vec<Arc<Mutex<ActiveShellCommand>>>);
+pub struct ActiveProcessMap(pub HashMap<Pid, ActiveProcess>);
 
 pub struct BevyLocalCommandsPlugin;
 
 impl Plugin for BevyLocalCommandsPlugin {
     fn build(&self, app: &mut App) {
-        app.add_event::<RunShellCommand>()
-            .add_event::<ShellCommandStarted>()
-            .add_event::<ShellCommandOutput>()
-            .add_event::<KillShellCommand>()
-            .add_event::<ShellCommandCompleted>()
-            .init_resource::<ActiveShellCommands>()
+        app.add_event::<RunProcess>()
+            .add_event::<ProcessStarted>()
+            .add_event::<ProcessOutputEvent>()
+            .add_event::<KillProcess>()
+            .add_event::<ProcessCompleted>()
```

Comment:
Don't we need to add `ProcessError` as an event here?

Reply:
Ha, nice catch. I'll add that later today.

Reply:
Added